### ISSUE TITLE
Arm-based instance type support

### ISSUE THREAD

**Issue by labisso** [aws/elastic-beanstalk-roadmap#4]
We are considering adding support for Arm based instance types including Graviton instances to Elastic Beanstalk platforms. Arm-based instances provide significant cost savings for compatible applications. Please let us know your interest in this potential improvement by reacting with a 👍 on the issue if you want it to be implemented. Additional feedback and comments also appreciated!

**Comment by hutm** [comment]
That would be huge cost savings for us. Is there any progress here?

**Comment by labisso** [comment]
@hutm We're still evaluating this feature. One followup question for you: are you most interested in A1 support for specific platforms?

**Comment by hutm** [comment]
java8

**Comment by johnthuss** [comment]
for Tomcat 8.5

**Comment by jonreiter** [comment]
single- and multi-container docker

**Comment by biblicabeebli** [comment]
Would love to have this. May be slower, but most of our stacks are architecture agnostic (Python), and the performance-price tradeoff here is probably an advantage when compared to T-series options.

**Comment by PUCHAJDA** [comment]
.NET on Windows Server with IIS. Could you estimate when this will be available for a1 with beanstalk instances ?

**Comment by tonyerskine** [comment]
Tomcat

**Comment by uldall** [comment]
Java 11 (Corretto)

**Comment by MikeZTM** [comment]
Corretto 8 with/without Tomcat

**Comment by tayl0r** [comment]
node.js

**Comment by G4brym** [comment]
python

**Comment by TonyFNZ** [comment]
It would be great to support M6g instance family (rather than A1)

**Comment by johndogg02** [comment]
PHP 7.3 / 7.4

**Comment by ueberhund** [comment]
Would love to see this request scaled out to support the new Graviton2 processor instance types

**Comment by ab77** [comment]
... especially `t4g's` for the tight.

**Comment by nkhedlekar** [comment]
Would love to see support for m6g and t4g in Beanstalk. This would give us significant cost savings as compared to current t3 - we run 100s of beanstalks in production !

**Comment by Tehnix** [comment]
Since Corretto is now available, would it make sense to at least enable the ARM instances (I'm personally looking for c6g.large and c6g.xlarge) for at least the Coretto platform?

**Comment by maewolfsky** [comment]
In general the delay of new instance type support coming to beanstalk should be shortened dramatically. A1s were added in 2018 and still probably wont have support until 2021?

**Comment by saarea** [comment]
Would really love to see ARM based instances to run net core 5.0 on amazon linux 2.  

We've already seen significant cost savings by moving to net core on linux from .net framework on windows.  ARM instances would be the cherry on the cake.

**Comment by 0x079** [comment]
> In general the delay of new instance type support coming to beanstalk should be shortened dramatically. A1s were added in 2018 and still probably wont have support until 2021?

Arm is clearly the future. I'm actually really surprised by how long this has taken, and how much longer we still have to wait... :-/

**Comment by lukaHorvatt** [comment]
Very interested to move 3 beanstalk environments from t3 to t4g as soon as it's available.

**Comment by LionelMarbot** [comment]
Is there any known timeline for the availability of t4g instances? Our t3 reserved instances have expired and we need to decide whether we should renew them or wait for t4g to become available.

**Comment by jordanade** [comment]
Any ETA on this?

**Comment by gsogol** [comment]
Any updates? 

node.js

**Comment by Bolandish** [comment]
@labisso Do you have an update? 👀

**Comment by Kraignos** [comment]
Any update?

**Comment by Novalaina** [comment]
Wow ... been a while with no new updates on this. Guess we can expect a long wait if ever before we can have the savings?

**Comment by jasontrumpio** [comment]
Would really love if this would finally be available...

**Comment by ariana-farshchi** [comment]
This would be an awesome feature to have!!

**Comment by jordanade** [comment]
Want PHP 8 on AL2 on t4g

**Comment by cpreese** [comment]
Tomcat 8.5 or greater + Corretto 11

**Comment by johnthuss** [comment]
The only reason this isn't available yet is because Amazon benefits financially from not making it available to beanstalk users. It's a tax on us. There is no technical reason they can't add easily add this instance to the list of supported instances.

**Comment by sprite2005** [comment]
Was going to try to see what changes I needed to make my run EB instances on m6g, but apparently it's not supported. Any ETA for this?

**Comment by lambrospetrou** [comment]
I am waiting for the Go platform to support ARM instance types...

**Comment by felixmeziere** [comment]
Great need on multi-container Docker please!

**Comment by mech** [comment]
Needed t4g and m6g for Ruby and Docker container for Beanstalk definitely.

**Comment by Nicoowr** [comment]
That would be awesome!

**Comment by Smotrov** [comment]
NodeJS on C6g 🥰

**Comment by ffMathy** [comment]
.NET Core here :heart_eyes:

**Comment by hwildwood** [comment]
We'd love to be able to use the new M6 instances for our JVM web app.

**Comment by AlanOuyang** [comment]
r6g.2xlarge supported?

**Comment by Smotrov** [comment]
> r6g.2xlarge supported?

Nothing "g" is supported in Beanstalk yet. Unfortunately.

**Comment by ffMathy** [comment]
Docker support would be nice as well! That would enable anyone to run on ARM.

**Comment by GavChap** [comment]
This would be of great interest to my organisation.

**Comment by fresnel-bruno** [comment]
Hi, any idea when the t4's will be supported? thanks!

**Comment by joswayski** [comment]
Bumping - Node :(

**Comment by Ricardonacif** [comment]
Needed t4g and m6g for Ruby and Docker container for Beanstalk definitely.

**Comment by snnles** [comment]
Hi All,

Thank you for your patience. We are in the process of launching Graviton2 support for Elastic Beanstalk. To better understand your needs, we have some questions for you. It will be helpful if you can answer below questions. Thank you for your responses in advance.

1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

3. How would you like to choose an instance type for your workload?

- I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.

- After selecting the processor, I would like to select an instance type out of a list of available options.

- After selecting the processor, I would like Elastic Beanstalk to choose the instance type for me based on my application (provide a default selection).

4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

**Comment by JustinTArthur** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

Python 3.7, 3.8 (hopefully 3.9 soon?) and Node.js (maintenance and active LTS versions)

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

Our Python and Node.js apps often don't involve binary executable workloads so those would not be concerning. It would be important that a C and C++ compiler toolchain be installed or installable on the AMIs for the sake of Python packages or node-gyp packages with machine code extensions where there is no arm64 wheel readily available.

> 3. How would you like to choose an instance type for your workload?

I don't have a strong preference here, but if we were to find that one of our dependencies does not build on one of the processor archs, then segregating by processor type first would make sense.

> 4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

As long as the graviton2 options were configurable in CloudFormation, it would suit our automation needs. All of the ASG spot instance percentage/allocation features that are exposed in Beanstalk would need to work with the Graviton2 workloads for it to be useful to us.

> 5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

This would be a luxury but not required for us to get started.

**Comment by dgagnon** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)  

Python, NodeJS, PHP

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?  

none that I can think of.

> 3. How would you like to choose an instance type for your workload?  

* I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.

> 1. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation,  
 configuration option, or tool that would help you migrate more easily?  

ideally, the same way we do platform upgrades.  (i.e. switching from node12 to node14 using the cli)

> 2. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)  

no need, Graviton2 is always cheaper.

**Comment by uldall** [comment]
>     1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

Java SE (Corretto 11), Docker
 
>     2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

Non.

>     3. How would you like to choose an instance type for your workload?

No preference.

>     4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

As long as there is support in CloudFormation then we are fine.

>     5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

This is not needed.

**Comment by waf** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

.NET 5 web applications (ASP.NET Core) running on Linux. Either nginx-based or Kestrel-based is fine.

> 2. Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?

We don't need .NET Framework or .NET Core 3.1. We're most interested in .NET 5, as it's had [a lot of excellent work done](https://devblogs.microsoft.com/dotnet/arm64-performance-in-net-5/) on ARM performance. Additionally, no Windows Server support required for us.

> 3. How would you like to choose an instance type for your workload?

No strong preference. Any of these three options are fine. We're most interested in compute-optimized instance types (C5/C6) and currently run on spot instances.

> 4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

We'd like to use cloudformation, `.ebextensions`, and platform hooks during our migration. As long as these EB features work, no other special automation or configuration is necessary.

> 5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

AWS Cost Explorer split by instance type would be nice, but not required.

**Comment by hwildwood** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

JVM Corretto 11 (Scala), Docker

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

None

> 3. How would you like to choose an instance type for your workload?

After selecting the processor, I would like to select an instance type out of a list of available options.

> 1. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

It would have to be a manual migration anyway as we'll need to rebuild a Docker image with a different JVM for ARM.

> 2. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Nice to have, not required.

**Comment by GavChap** [comment]
>What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

PHP, Multi-container docker

> Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?

None

> How would you like to choose an instance type for your workload?

After selecting the processor, I would like to select an instance type out of a list of available options, or you could select the processor type based on the instance, just a flat list of instance types would also be acceptable, it's implied that graviton instances are ARM anyway.

> How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

As long we can configure the instance types and processor type in cloudformation we'll be ok. We'd be using docker multi-arch images anyway.

> Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Something in cost Explorer would be good.

**Comment by fresnel-bruno** [comment]
Great news!
1: We would like to run Docker apps
2: none
3: no preference
4: I don't think we need anything special, as long it supports AL2 (with cloudwatch agent)
5: we will measure the performance using our test suite, I don't think we need anything from the platform there.

Many thanks!

**Comment by jasontrumpio** [comment]
1) Django/Python 3.7/3.9

2) N/A

3) I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.

4) We'd leverage our automation to redeploy the applications.

5) Cost Explorer would be cool, but not a priority.

**Comment by TheObtuseAutodidact** [comment]
`What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)`
Currently running Docker running Java

`Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?`
My hope is that because we're running docker this is an easy swap. 


`How would you like to choose an instance type for your workload?`

* I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.
This seems like the way it works now. Fine by me. 

`How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?`
No special requests

`Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)`
That would be a great "nice to have"  feature.

**Comment by johnthuss** [comment]
> Hi All,
> 
> Thank you for your patience. We are in the process of launching Graviton2 support for Elastic Beanstalk. To better understand your needs, we have some questions for you. It will be helpful if you can answer below questions. Thank you for your responses in advance.
> 
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

Tomcat/Corretto 11

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

None

> 3. How would you like to choose an instance type for your workload?
> 
> * I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.
> * After selecting the processor, I would like to select an instance type out of a list of available options.
> * After selecting the processor, I would like Elastic Beanstalk to choose the instance type for me based on my application (provide a default selection).

A flat list is fine with me.

> 
> 1. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

Just select a different instance type from the list in the Configuration area.

> 2. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

I don't need metrics specific to me as long as general comparison data is available.

**Comment by gsogol** [comment]
1. node.js (latest)

2. Will web sockets still work over nginx? We configure nginx to allow it currently 

3. we use .ebextensions and we specify the type there during our deployment

4. We'd leverage our automation to redeploy the applications.

5. Cost Explorer is not a priority

**Comment by ThomasWie** [comment]
Hi,

> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)
.NET 5 web applications on Linux

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?
N/A


> 3. How would you like to choose an instance type for your workload?
> * After selecting the processor, I would like to select an instance type out of a list of available options.
> 
> 1. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

I would just change the instance type, via script or manually.

> 2. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

AWS Cost Explorer is fine

BR,

Thomas

**Comment by daysofwineandroses** [comment]
>     1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

- Tomcat 8.5 with Corretto 8 running on 64bit Amazon Linux 2
- Tomcat 8.5 with Corretto 11 running on 64bit Amazon Linux 2
- Corretto 11 running on 64bit 

>     2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?
None 
>     3. How would you like to choose an instance type for your workload?
Needs CloudFormation support, we currently have 15 Beanstalk environments deployed via CloudFormation. 
>     4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?
We'd just update our CloudFormation template and stack parameters. Our application WARs should just run.
>     5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)
More CloudWatch metrics are always nice, not sure what that would look like though.

**Comment by dercoder** [comment]
> 1. What types of applications would you like to run on Graviton2? 

PHP7 / PHP8

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?
If depends if the required PHP extensions will support this processor architecture

> 3. How would you like to choose an instance type for your workload?

After selecting the processor, I would like to select an instance type out of a list of available options.

> 4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

I think it always depends on the language. PHP does not need be recompiled or somehing. 
Only change the instance type and reinstall the system. That's it. PHP extensions will be also installed by scripts for the specific architecture.

>5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance?

That would be nice. 
I would prefere to see how much money could be saved by switching to Graviton2 with the same or better performance. 
Like: switch to Graviton2 Instace XX and you will get +15% performance and save -10% on your monthly bill.

**Comment by matthewdouglas** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

Python 3.8+, possibly .NET 5+ in the future.

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

Our .NET Framework applications running on Windows Server 2016. Probably best to migrate to .NET 5 first :)

> 3. How would you like to choose an instance type for your workload?

* I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.
This is not a strong preference on console UI. Need to support via CloudFormation and .ebextensions.

> 4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

No immediate requests here.

> 5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Nice to have built-in features around this, but not necessary.

**Comment by coffeeburrito** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

PHP 8, Corretto 11

> 2. Which applications would you _not_ want to run on Graviton2? What are your concerns with migrating these applications?

My only concern is that we might find some niche use case that performs poorly, but as long as we can easily switch between node types then it's not a problem.

> 3. How would you like to choose an instance type for your workload?

Doesn't matter; using CloudFormation

> 1. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

Since I use named environments, it would be nice if switching t4g type doesn't require environment replacement.

> 2. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Yes, I'd probably use before vs after comparisons of CloudWatch data. If you want a "fun" idea, let people switch half the nodes in a load balanced environment to t4g and do a side-by-side performance comparison using live data.

**Comment by thiagoaraujogit** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

NodeJS 12/14

> 2. Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?

I don't have a type of application that I would NOT want to run, the only concern is about compatibility 
related to the ARM architecture. But as long as it goes through a good testing phase, everything will be fine :)

> 3. How would you like to choose an instance type for your workload?

An list of instance types at Elastic Beanstalk UI, including the ARM based instances. Or support via Cloudformation and .ebextensions folder.

> 4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

No immediate requests for that, I just use the Immutable Deployment with the new configuration.

> 5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Yes, measuring performance would be very good to present numbers and benefits. I would use AWS Cost Explorer and Cloudwatch.

**Comment by hayd** [comment]
Does AWS not have any infrastructure for sending out surveys? This seems a ridiculous way to do it! 😆 

---

1. nginx + python + corretto8 tomcat
2. 🤷 
3. Trial and error / load testing. Just like any other eb:  I'd select from a list/set in a config file.
4. If it were a new and **distinct platform type**, then I would be creating a new application, and deleting the old once I have the new working. I don't see why any migration aid would be necessary. Having different instance types in the same application platform having incompatible binaries seems a recipe for disaster. 
5. Sure. 🤷

**Comment by cpreese** [comment]
What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

      Tomcat 8.5 and higher, Corretto 11 and higher on Amazon Linux 2

    Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?

     None

    How would you like to choose an instance type for your workload?

    Cost basis for instance that meets memory requirements currently 8 GB RAM 

    I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.

    Flat list works for us 

    After selecting the processor, I would like to select an instance type out of a list of available options.

    PROD Based on cost, non PROD environments we would select a group for SPOT 

    After selecting the processor, I would like Elastic Beanstalk to choose the instance type for me based on my application (provide a default selection).

    How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

   We would create a new environment using CloudFormation and switch Route53 CNAME record

    Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

    Cost of a instance with sufficient memory currently 8 GB

**Comment by jpike88** [comment]
> What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

Node 12+

> Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?

None

> How would you like to choose an instance type for your workload?

I would like to select an instance type out of a flat list of x86 + Graviton2 instance types.

> How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

I guess I'll be doing some sort of clone + swap environment as I imagine my stack won't just work on ARM out the gate.

> Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Cost Explorer works fine for me

**Comment by pp-assis** [comment]
> 1. What types of applications would you like to run on Graviton2? (For example: Python 3.7, PHP7)

NodeJS 12/14

> 2. Which applications would you not want to run on Graviton2? What are your concerns with migrating these applications?

I don't have a type of application that I would NOT want to run, the only concern is about compatibility
related to the ARM architecture. But as long as it goes through a good testing phase, everything will be fine :)

> 3. How would you like to choose an instance type for your workload?

An list of instance types at Elastic Beanstalk UI, including the ARM based instances. Or support via Cloudformation and .ebextensions folder.

> 4. How would you like to migrate your existing workload to Graviton2? Can Elastic Beanstalk add any automation, configuration option, or tool that would help you migrate more easily?

No immediate requests for that, I just use the Immutable Deployment with the new configuration.

> 5. Would you like to be able to measure or compare the price-performance ratio of your application on Graviton2 vs. x86? If you would, how would you measure price-performance? (For example: AWS Cost Explorer, Amazon CloudWatch)

Yes, measuring performance would be very good to present numbers and benefits. I would use AWS Cost Explorer and Cloudwatch.

**Comment by jpike88** [comment]
As a side note, I'd much prefer if these became available before the old amazon linux 1 environments become retired. I currently don't see much point of transitioning our environments to Amazon Linux 2 x64 if I can wait and just jump straight to ARM.

**Comment by thiagoaraujogit** [comment]
Any updates?

**Comment by antgel** [comment]
> Any updates?

Yes, you bumped the thread with a pointless question with an obvious answer and 58 people got a pointless ping (yes I'm aware of the irony of sending this). I'd suggest adding your :+1: to the OP on the reasonable assumption that AWS are using those to gauge community feedback.

**Comment by 1tonyca** [comment]
For general comments and feedback, please continue to use this thread.

To track rollout updates, please follow https://github.com/aws/elastic-beanstalk-roadmap/issues/192

**Comment by mometic** [comment]
Sure would be nice to get a regular update on this.  Elasticbeanstalk has been the laggard in modern processor support for some time.  Many are making decisions about switching architectures as this is outdated as 40% ARM efficiencies are tangible.

AWS - how about some meaningful release dates?

**Comment by snnles** [comment]
arm64-based instance type is now supported by Elastic Beanstalk. Closing this thread.

To track more updates related to arm64-based instance type support on Elastic Beanstalk console, please follow [issue #212](https://github.com/aws/elastic-beanstalk-roadmap/issues/212)
